GameServerCore/Domain/GameObjects/IAttackableUnit.cs

```diff
@@ -1,22 +1,25 @@
-﻿using GameServerCore.Enums;
-
-namespace GameServerCore.Domain.GameObjects
-{
-    public interface IAttackableUnit : IGameObject
-    {
-        bool IsDead { get; }
-        bool IsModelUpdated { get; set; }
-        int KillDeathCounter { get; set; }
-        int MinionCounter { get; }
-        string Model { get; }
-        IReplication Replication { get; }
-        IStats Stats { get; }
-
-        void Die(IAttackableUnit killer);
-        void TakeDamage(IAttackableUnit attacker, float damage, DamageType type, DamageSource source, bool isCrit);
-        void TakeDamage(IAttackableUnit attacker, float damage, DamageType type, DamageSource source, DamageText damageText);
-
-        bool IsInDistress();
-        bool ChangeModel(string model);
-    }
-}
+﻿using GameServerCore.Enums;
+
+namespace GameServerCore.Domain.GameObjects
+{
+    public interface IAttackableUnit : IGameObject
+    {
+        bool IsDead { get; }
+        bool IsModelUpdated { get; set; }
+        int KillDeathCounter { get; set; }
+        int MinionCounter { get; }
+        string Model { get; }
+        IReplication Replication { get; }
+        IStats Stats { get; }
+        void Die(IAttackableUnit killer);
+        int[] ShieldAmount { get; }
+        
+        void Die(IAttackableUnit killer);

```

Comment:
You have 2 definitions of the same thing. Should change (the build now fails because of this).

Reply:
Done.